rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -894,6 +897,26 @@ stream {
         location = {{ $authPath }} {
             internal;
 
+            {{ if $externalAuth.AuthCacheKey }}
+            set $cache_key '{{ $externalAuth.AuthCacheKey }}';
+
+            rewrite_by_lua_block {
+                local hash = ngx.sha1_bin(ngx.var.cache_key)
+                ngx.var.cache_key = ngx.encode_base64(hash)
```

Comment:
I wonder if we should introduce a new variable. You are overloading the variable with two different type of values.

Reply:
i'm mutating the value, yes. but the type stays from my understanding the same (string).

Adding an extra variable makes it unreadable IMO because we have to declare the variable beforehand:

```
set $tmp_cache_key '{{ ..things... }}';
set $real_cache_key ''; # <--- this

rewrite_by_lua {
  ngx.var.real_cache_key = nginx.encode_base64(ngx.sha1_bin(ngx.var.tmp_cache_key))
}

proxy_cache_key "$real_cache_key";

```

Reply:
I don't feel strongly, feel free to not change this.

But subjectively I like the more verbose version you posted above.

Reply:
i changed to the slightly more verbose version.